expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt

```diff
@@ -0,0 +1,107 @@
+package com.expediagroup.sdk.rest.util
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import java.net.URLDecoder
+import java.nio.charset.StandardCharsets
+
+/**
+ * Parses URL path template parameters and returns a map of template segments to path segments.
+ *
+ * @param path The actual URL path.
+ * @param template The URL template with placeholders.
+ * @return A map where the keys are template segments and the values are the corresponding path segments.
+ * @throws IllegalArgumentException if either path or template is null.
+ */
+fun parseUrlPathTemplateParams(
+    template: String,
+    path: String
+): Map<String, String> =
+    listOf(template, path).map {
+        it.removePrefix("/")
+            .removeSuffix("/")
+            .split("/")
+            .map(String::trim)
+    }.also {
+        val templateSegments = it[0]
+        val pathSegments = it[1]
+
+        require(templateSegments.size == pathSegments.size) {
+            "Template and path segments must have the same number of segments"
+        }
+    }.zipWithNext { templateSegment, pathSegment ->
+        templateSegment.zip(pathSegment)
+    }.flatten().filter { (key, value) ->
+        key != value && key.startsWith("{") && key.endsWith("}")
+    }.associate { (key, value) ->
+        key.removePrefix("{").removeSuffix("}") to value
+    }
+
+/**
+ * Parses URL query parameters from a query string and returns a map of parameter names to lists of values.
+ *
+ * @param query The URL query string.
+ * @param destringifyStrategies The strategy to destringify query parameter values (e.g., comma-delimited, space-delimited).
+ * @return A map where the keys are parameter names and the values are lists of parameter values.
+ */
+fun parseUrlQueryParamsFromQuery(
```

Comment:
If we move these functions to `UrlUtil` or `UrlHelper` class, we can simplify the name to
```suggestion
fun parseQueryParams(
```

Reply:
Done.